diff --git a/dist/cjs/types.d.cts b/dist/cjs/types.d.cts
new file mode 100644
index 0000000000000000000000000000000000000000..e6f3462c7fee570a87e8c7b9a3ea1ca8549ca89a
--- /dev/null
+++ b/dist/cjs/types.d.cts
@@ -0,0 +1,13 @@
+import type { ConfigObject } from "@eslint/core";
+
+export type Config = ConfigObject;
+export type InfiniteArray<T> = T | InfiniteArray<T>[];
+export type ExtensionConfigObject = Omit<ConfigObject, "basePath"> & {
+	extends?: never;
+};
+export type SimpleExtendsElement = string | ExtensionConfigObject;
+export type ExtendsElement = SimpleExtendsElement | InfiniteArray<ExtensionConfigObject>;
+export interface ConfigWithExtends extends ConfigObject {
+	extends?: ExtendsElement[];
+}
+export type ConfigWithExtendsArray = InfiniteArray<ConfigWithExtends>[];
