diff --git a/lib/eslint/eslint.js b/lib/eslint/eslint.js
index c6a358d918fc8f9feca2b0f64a7fea7242fdf8ea..f26c662ed91d1d1cb3d839a9baa0ab1a82b1d827 100644
--- a/lib/eslint/eslint.js
+++ b/lib/eslint/eslint.js
@@ -18,6 +18,12 @@ const { SHARE_ENV, Worker } = require("node:worker_threads");
 const { version } = require("../../package.json");
 const { defaultConfig } = require("../config/default-config");
 const timing = require("../linter/timing");
+const builtInFormatters = {
+	stylish: require("../cli-engine/formatters/stylish"),
+	html: require("../cli-engine/formatters/html"),
+	json: require("../cli-engine/formatters/json"),
+	"json-with-metadata": require("../cli-engine/formatters/json-with-metadata"),
+};
 
 const {
 	createDebug,
@@ -1270,7 +1276,8 @@ class ESLint {
 		let formatter;
 
 		try {
-			formatter = (await import(pathToFileURL(formatterPath))).default;
+			formatter = (isBuiltInFormatter && builtInFormatters[normalizedFormatName]) ||
+				(await import(pathToFileURL(formatterPath))).default;
 		} catch (ex) {
 			// check for formatters that have been removed
 			if (isBuiltInFormatter && removedFormatters.has(name)) {
